### DISCUSSION TITLE
Componentes de JinjaX con extensión html

### DISCUSSION THREAD

**Discussion by jpsca** [jpsca/jinjax#10] [Preguntas/Respuestas] [answered]
@v4ld3rr4m4 [preguntó](https://github.com/jpsca/jinjax/discussions/9#discussion-4935939)  “se puede usar archivos html como componentes de JinjaX”

Respondo aquí en español

**Comment by jpsca** [comment, marked as answer]
Solo puedes usar archivos que estén en los folders que haya agregado usando `catalog.add_folder(ruta/del/folder/)`

En ese/esos folders `catalog.render(“Nombre”)`, JinjaX buscará un archivo que empieze con “Nombre” y termine en “.jinja”, así que puede tener más extensiones, por ejemplo “Nombre.html.jinja”.

Pero si quieres evitar la extensión “.jinja” completamente, puedes usar el argumento `__file_ext` al llamar a `catalog.render(…)`

Por ejemplo, este código:

```python
catalog.render("Producto", __file_ext="html")
```

…buscará un archivo “Producto.html” en vez de “Producto.jinja”

Alternativamente, puedes cambiar la extensión por defecto (o sea para TODOS los componentes), con el argumento `file_ext` al crear el catálogo:

```python
catalog = Catalog(…, file_ext="html")
```

**Reply by jpsca** [reply]
Los componentes **son** archivos HTML con una extensión diferente, y que además pueden contener código Jinja

**Reply by v4ld3rr4m4** [reply]
Gracias primero que todo. Juan. cierto un poco ambigua me quedo la pregunta.  El escenario es ya un proyecto existente. Y los html que me refiero son los clasicos layouts. con elemeentos BLOCK      END BLOCK. etc.   Y es en estos donde pretendia incluir mis nuevos componentes.   
Una posibilidad de exito segun lo que me cuentas seria cambiar los **render_template(---** por  **catalog.render("....**

**Reply by v4ld3rr4m4** [reply]
Fantastic. Success.   

As Mentioned  , in my case using flask , I changed render_template to catalog.render . Previous --> catalog.add_folder("templates")  and in Component Name parameter use htmlpagename . Of Course I need to pass all context vars and include it at the first line of htmlpagename {# def   vars,...    #}

**Reply by jpsca** [reply]
Update: Use `catalog.irender(...)` to render subcomponents instead of `catalog.render(...)`, because the later reset the collected css/js files
